src/Agent.Listener/Agent.cs

```diff
@@ -35,10 +34,16 @@ public override void Initialize(IHostContext hostContext)
 
         public async Task<int> ExecuteCommand(CommandSettings command)
         {
+            AssemblyLoadContext loadContext = AssemblyLoadContext.GetLoadContext(typeof(Agent).GetTypeInfo().Assembly);
             try
             {
                 _inConfigStage = true;
+                _completedCommand.Reset();
                 _term.CancelKeyPress += CtrlCHandler;
+
+                //register a SIGTERM handler
+                loadContext.Unloading += Agent_Unloading;
```

Comment:
There is a comment in AgentService.js `// TODO wait for 30 seconds and send a SIGKILL`. Is any change there required as part of this PR?